# Graded assignment 1

Nick Weerts
U1265916

# Packages

We need the package `tidyverse`

In [19]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [20]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1         V2         V3         V4         V5         V6        V7       
1 0.01517902 0.06795832 0.08337714 0.02140359 0.97439022 0.4552703 0.9359242
2 0.73665159 0.54593688 0.99901210 0.39842068 0.09180459 0.2270290 0.5873688
  V8        V9        V10        V11       V12        V13       V14      
1 0.1877911 0.6053422 0.36573702 0.8104265 0.28604657 0.9275326 0.1878352
2 0.4185559 0.7897977 0.08103879 0.9796341 0.07776791 0.8757742 0.3040303
  V15       V16       V17         V18       V19       V20      
1 0.2203071 0.7534879 0.245819874 0.6680065 0.9401852 0.7440187
2 0.9473032 0.5089313 0.003466451 0.2959484 0.2580756 0.3454929

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [21]:
df2 <- gather(df1, Key, Value, V1:V20)
str(df2)
head(df2)

'data.frame':	40 obs. of  2 variables:
 $ Key  : chr  "V1" "V1" "V2" "V2" ...
 $ Value: num  0.0152 0.7367 0.068 0.5459 0.0834 ...


Key Value     
1 V1  0.01517902
2 V1  0.73665159
3 V2  0.06795832
4 V2  0.54593688
5 V3  0.08337714
6 V3  0.99901210

## Assignment 2

We will generate some data:

In [22]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [23]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [29]:
df5 <- merge(df3, df4, all=TRUE)
head(df5)

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook